File: python/send_message/functions/send_email_mailgun.py
import requests
import os


def send_email_mailgun(variables, email, message, subject):
    """Send email using Mailgun"""

    if not email or not message or not subject:
        raise Exception("Missing email, message or subject")

    domain = variables.get("MAILGUN_DOMAIN", None)
    api_key = variables.get("MAILGUN_API_KEY", None)

    if not domain:
        raise Exception("Missing Mailgun domain")
    if not api_key:
        raise Exception("Missing Mailgun API key")

    try:
        response = requests.post(
            f"https://api.mailgun.net/v3/{domain}/messages",
            auth=("api", api_key),
            data={
                "from": "<welcome@my-awesome-app.io>",
                "to": email,
                "subject": subject,
                "text": message,
            },
        )
        # response.raise_for_status()

    except Exception as e:
        print(e)
        return {"success": False, "error": str(e)}

    return {
        "success": True,
    }

File: kotlin/sendMessageHelper/sendEmailMailgun.kt
import java.util.Base64

fun sendEmailMailgun(variables: Map<String, String>, email: String?, message: String?, subject: String?): Map<String, Any>{
    try {
        if (email.isNullOrEmpty() || message.isNullOrEmpty() || subject.isNullOrEmpty()){
            throw Exception("Missing email, message, or subject")
        }
    } catch (e: Exception) {
        println(e.message)
        return mapOf("success" to false,
                     "message" to "sendEmailMailgun failed")
    }

    val domain = variables["MAILGUN_DOMAIN"]
    val apiKey = variables["MAILGUN_API_KEY"]

    try {
        if (domain.isNullOrEmpty()){
            throw Exception("Missing Mailgun domain")
        }
    } catch (e: Exception) {
        println(e.message)
        return mapOf("success" to false,
                     "message" to "sendEmailMailgun failed")
    }

    try {
        if (apiKey.isNullOrEmpty()){
            throw Exception("Missing Mailgun API key")
        }
    } catch (e: Exception) {
        println(e.message)
        return mapOf("success" to false,
                     "message" to "sendEmailMailgun failed")
    }

    try {
        // URL for the Mailgun API endpoint
        val url = URL({"https://api.mailgun.net/v3/$domain/messages"})

        // authentication
        val auth = "api:$apiKey"

        // initializing data to send
        val email_data =  "{\"from\":\"$domain\", \"to\":\"$email\", \"subject\":\"$subject\", \"text\":\"$message\"}"

        // opening the HTTP connection    
        val connection = url.openConnection() as HttpURLConnection
        connection.doOutput = true
        connection.requestMethod = "POST"
        // setting request headers + authenticating the HTTP connection
        connection.addRequestProperty("Authorization", "Basic " + Base64.getEncoder().encodeToString(auth.toByteArray()))
        connection.addRequestProperty("Content-Type", "application/json")
        
        // make POST request
        val outputStream: OutputStream = connection.outputStream
        outputStream.write(email_data.toByteArray(Charsets.UTF_8))
        outputStream.flush()

        // check status of response
        val responseCode = connection.responseCode
        if (responseCode != HttpURLConnection.HTTP_OK) {
            throw Exception("$responseCode")
        }
        
        // housekeeping ; closing the open connection
        outputSream.close()
        connection.disconnect()
    } catch (e: Exception) {
        println(e.message)
        return mapOf("success" to false,
                     "message" to "sendEmailMailgun failed")
    }
    
    return mapOf("success" to true,
                 "message" to "You called sendEmailMailgun")
}